Cython/Compiler/TreePath.py

```diff
@@ -168,7 +168,7 @@ def select(result):
                 if attr_value == value:
                     yield attr_value
                 elif (isinstance(attr_value, bytes) and isinstance(value, str) and
-                        attr_value.decode("utf-8") == value):
+                        attr_value == value.encode()):
```

Comment:
This is also not going to work correctly in Py2, where `str` is `bytes` and `.encode()` will `.decode()` it first. If the `value` has some encoding other than `us-ascii`, then this implicit decoding will fail. Thus my suggestion that this should only be done for unicode strings and not Py2 `str`.

Reply:
Fixed (I think)